### PR TITLE
Create a new page about experimental features

### PR DESCRIPTION
**Pull request by kgolubic**
### Description
 
- Created a new page about experimental features.
- Added links to the new page.

### Pull request type

Please check what kind of PR this is:

- [ ] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
(especially necessary if the PR is related to a release)
https://github.com/memgraph/documentation/issues/812

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [ ] If release-related, add a product and version label
- [ ] If release-related, add release note on product PR

### CODE REVIEW

**Review by hal-eisen-MG — CHANGES_REQUESTED** [review]
Left several comments

**Comment by gitbuda** [comment]
This seems ok from the product perspective 😄 I think the statuses should be more accurate but we'll provide/update that before each release 😄

File: pages/database-management/experimental-features.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- **Status**: Experimental. Enhances data replication capabilities with added security and multi-tenancy features.
+- **Usage**: Enable the feature using the flag `--experimental-enabled=system-replication`.
+- **Documentation**: For more details, visit the [Experimental Replication documentation](/clustering/replication/experimental-replication).
+
+### Text search
+- **Description**: This feature enables text search capabilities within Memgraph. Text search only supports some usage modalities and is not guaranteed to work correctly in use cases involving concurrent transactions and replication.
+- **Status**: Experimental. Provides text search functionalities but is not guaranteed to work correctly in use cases that involve concurrent transactions and replication.
+- **Usage**: Enable the feature using the flag `--experimental-enabled=text-search`.
+- **Documentation**: For more details, visit the [Text Search documentation](/querying/text-search).
+
```
[the remark this answers is not in the document]
Comment (kgolubic) [reply pages/database-management/experimental-features.mdx:35]: I'll add a note about that.

File: pages/database-management/experimental-features.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+title: Experimental features
+description: Experimental features are new functionalities in Memgraph that carry some risk. They are still in development and may have stability issues.
```
Comment (hal-eisen-MG) [inline pages/database-management/experimental-features.mdx:3]: Please add a phrase along the lines of "contact your support representative when encountering any challenges and we'll respond promptly to get you back on track quickly"

File: pages/database-management/experimental-features.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+If you already have a Memgraph instance with an experimental feature enabled,
+and you want to add another one, you will need to restart Memgraph. Ensure that
+all the necessary flags are enabled during the restart.
+
+</Callout>
+
+## Experimental features list
+
+### High availability mode
+- **Description**: This feature enables high availability capabilities in Memgraph Enterprise Edition.
```
Comment (hal-eisen-MG) [inline pages/database-management/experimental-features.mdx:35]: Avoid defining a term with itself: "high availability" in this case. I'd recommend something about "resilience" and/or "fault tolerance" here.

File: pages/database-management/experimental-features.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- **Status**: Experimental. Designed to improve system reliability and uptime, but may still have stability issues.
+- **Usage**: Enable the feature using the flag `--experimental-enabled=high-availability`.
+- **Documentation**: For more details, visit the [High Availability Mode documentation](/database-management/configuration).
+
+### System replication
+- **Description**: This feature enables authentication and multi-tenancy replication in Memgraph Enterprise Edition.
```
Comment (hal-eisen-MG) [inline pages/database-management/experimental-features.mdx:41]: Avoid using a term to define itself. In this case, "replication" should be referring to headers and followers, along with a mention of the choice between sync and async, and perhaps a mention of horizontal scalability.

File: pages/database-management/experimental-features.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- **Status**: Experimental. Enhances data replication capabilities with added security and multi-tenancy features.
+- **Usage**: Enable the feature using the flag `--experimental-enabled=system-replication`.
+- **Documentation**: For more details, visit the [Experimental Replication documentation](/clustering/replication/experimental-replication).
+
+### Text search
+- **Description**: This feature enables text search capabilities within Memgraph. Text search only supports some usage modalities and is not guaranteed to work correctly in use cases involving concurrent transactions and replication.
```
Comment (hal-eisen-MG) [inline pages/database-management/experimental-features.mdx:47]: Avoid defining a term with itself. In this case, "text search" should explain the core concept of finding nodes based on property values, and mention the choice between literal and regex, and the aggregation concept.

**Review by hal-eisen-MG — APPROVED** [review]
LGTM

**Pull request merged**